pkg/operator/genericoperatorclient/dynamic_operator_client.go

```diff
@@ -170,6 +174,91 @@ func (c dynamicOperatorClient) UpdateOperatorStatus(ctx context.Context, resourc
 	return retStatus, nil
 }
 
+func (c dynamicOperatorClient) ApplyOperator(ctx context.Context, fieldManager string, applyConfiguration *applyoperatorv1.OperatorSpecApplyConfiguration) (err error) {
+	applyMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(applyConfiguration)
+	if err != nil {
+		return fmt.Errorf("failed to convert to unstructured: %w", err)
+	}
+	applyUnstructured := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"spec": applyMap,
+		},
+	}
+	applyUnstructured.SetGroupVersionKind(c.kind)
+	applyUnstructured.SetName(c.configName)
+
+	uncastOriginal, err := c.informer.Lister().Get(c.configName)
+	if err != nil {
+		return fmt.Errorf("unable to read existing: %w", err)
+	}
+	switch {
+	case apierrors.IsNotFound(err):
+		// do nothing and proceed with the apply
+	case err != nil:
+		return fmt.Errorf("unable to read existing: %w", err)
+	default:
+		original := uncastOriginal.(*unstructured.Unstructured)
+		if allRequiredFieldsArePresent(original, applyUnstructured) {
```

Comment:
not sure about this, because that would imply that we would have to change all controllers to do the same. as of today there is no built-in method into client-go that would provide such a functionality.

Reply:
renamed to make it more clear.